speedtest/test_performance_openssl.py

```diff
@@ -0,0 +1,247 @@
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        result = "OK"
+        if exc_val is not None:
+            result = "ERROR"
+        print "\n..end {} {}.. ({}sec)".format(self.message, result,
+                                               time.time() - self.started)
+
+
+def download_openssl():
+    """
+    Get the openssl zip and unpack it
+    """
+    if not os.path.exists(OPENSSL_ZIP):
+        getoslsrc = urllib.URLopener()
+        with BlockMessage("download openssl"):
+            getoslsrc.retrieve(OPENSSL_URL, OPENSSL_ZIP + ".part")
+            os.rename(OPENSSL_ZIP + ".part", OPENSSL_ZIP)
+
+
+def clean_openssl_build():
+    """
+    Unpack the openssl source, possibly deleting the previous one
+    :return:
+    """
+    with zipfile.ZipFile(OPENSSL_ZIP, "r") as unzip:
+        folder = unzip.namelist()[0]
+        if os.path.exists(folder):
+            with BlockMessage("delete old openssl folder"):
+                retry_delete(folder)
+
+        with BlockMessage("unzip openssl"):
+            unzip.extractall()
+
+        if len(SOURCES) == 0:
+            SOURCES.append(folder.rstrip("/"))
+
+
+def find_visual_studio():
+    """
+    Attempt to find vs 11 or vs 12
+    :return:
+    """
+    vcvers = ["13.0", "12.0", "11.0"]
+    for vc in vcvers:
+        vcdir = os.path.join("c:\\", "Program Files (x86)",
+                             "Microsoft Visual Studio {}".format(vc),
+                             "VC", "bin")
+        vcvars = os.path.join(vcdir, "vcvars32.bat")
+        if os.path.exists(vcvars):
+            return vcdir, vcvars
+
+    raise Exception("cannot find visual studio!")
+
+
+def configure_openssl():
+    """
+    Run the configure steps (requires perl)
+    :return:
+    """
+    with BlockMessage("configure openssl"):
+        subprocess.check_call(["perl",
+                               "Configure", "VC-WIN32", "no-asm", "--prefix=c:\openssl"],
+                              env=ENVS,
+                              cwd=SOURCES[0])
+
+    with BlockMessage("generate makefiles"):
+        subprocess.check_call([os.path.join("ms", "do_ms.bat")],
+                              shell=True,
+                              env=ENVS,
+                              cwd=SOURCES[0])
+
+
+def setup_function(request):
+    """
+    Ensure a clean build tree before each test
+    :return:
+    """
+    clean_openssl_build()
+    configure_openssl()
+
+
+def setup_module():
+    """
+    Check that our exe has been built.
+    :return:
+    """
+    if not os.path.isfile(CLCACHE):
+        pytest.fail("please build the exe first")
+
+    _, vcvars = find_visual_studio()
+    with BlockMessage("getting vc envs"):
+        getenvs = subprocess.check_output([os.path.join(THISDIR, "get_vc_envs.bat"), vcvars])
+        for line in getenvs.splitlines():
+            if "=" in line:
+                name, val = line.split("=", 1)
+                ENVS[name.upper()] = val
```

Comment:
I think you could drop the `get_vc_envs.bat` file in favor of just doing:

```python
output = subprocess.check_output([vcvars, '>', 'NUL', '&&', 'set'], shell=True)
ENVS = dict(line.split('=', 1) for line in output.splitlines())
```

I.e. just run the batch file and then run `set` to print the environment, then parse that. This clearly assumes that every line in the `set` output has at least one `=` sign -- I hope that's true.